src/rgw/rgw_rados.h

```diff
@@ -3557,7 +3557,7 @@ class RGWRados
   }
 
   bool need_to_log_metadata() {
-    return get_zone().log_meta;
+    return is_meta_master() && get_zone().log_meta;
```

Comment:
correct, and `RGWZoneGroup::post_process_params()` has already been enforcing this in `radosgw-admin` commands:
```c++
  for (map<string, RGWZone>::iterator iter = zones.begin(); iter != zones.end(); ++iter) {
    RGWZone& zone = iter->second;
    zone.log_data = log_data;
    zone.log_meta = (is_master && zone.id == master_zone);
```
this pr just moved that enforcement into `RGWRados` itself, so that we can properly handle zones that were created before jewel and hadn't triggered that code in `RGWZoneGroup::post_process_params()`

after jewel, if multiple zones have `log_meta` set, the secondary zone will add each mdlog entry that it syncs from the metadata master back into its mdlog, and then keep resyncing/rewriting that entry, filling up its mdlog with duplicate entries